### PR TITLE
Fix race condition in g_pfnGetSystemTimeAsFileTime initialization

### PR DESCRIPTION
**Pull request by jkotas**
Multiple threads initializing g_pfnGetSystemTimeAsFileTime at the same time can end up with different conclusions. Make the result consistent.

### CODE REVIEW

**Comment by jkotas** [comment]
https://github.com/dotnet/coreclr/pull/22383#issuecomment-461443940 for context

cc @filipnavara

File: src/classlibnative/bcltype/system.cpp [hunk 1]
```diff
@@ -72,23 +72,28 @@ void WINAPI InitializeGetSystemTimeAsFileTime(LPFILETIME lpSystemTimeAsFileTime)
...
     if (func == NULL)
-#endif
     {
         func = &::GetSystemTimeAsFileTime;
     }
 
-    g_pfnGetSystemTimeAsFileTime = func;
+    InterlockedCompareExchangeT(&g_pfnGetSystemTimeAsFileTime, func, &InitializeGetSystemTimeAsFileTime);
+
     func(lpSystemTimeAsFileTime);
```
Comment (filipnavara) [inline src/classlibnative/bcltype/system.cpp]: This should call g_pfnGetSystemTimeAsFileTime or the race condition can still happen.

**Comment by filipnavara** [comment]
Thanks for addressing it!

File: src/classlibnative/bcltype/system.cpp (hunk 1 above)
Comment (jkotas) [reply src/classlibnative/bcltype/system.cpp]: good catch. Thanks!

**Pull request merged**